packages/core/lib/commands/run/run.js

```diff
@@ -1,61 +1,33 @@
 const TruffleError = require("@truffle/error");
+const { Plugin } = require("@truffle/plugins");
 
 const Run = {
   // initiates running the third-party command
-  initializeCommand(pluginConfigs, customCommand) {
-    let pluginCommandObj = {};
-
-    for (let plugin in pluginConfigs) {
-      try {
-        if (pluginConfigs[plugin].commands[customCommand])
-          pluginCommandObj[plugin] =
-            pluginConfigs[plugin].commands[customCommand];
-      } catch (_) {
-        // not added to pluginCommandObj
-        // empty pluginCommandObj handled in checkPluginObject
+  initializeCommand(customCommand, plugins) {
+    // look for plugin defining customCommand
+    for (const plugin of plugins) {
+      if (plugin.definesCommand(customCommand)) {
+        try {
+          return plugin.loadCommand(customCommand);
+        } catch (_) {
+          // loading fails; try next one
+        }
       }
     }
-    this.checkPluginObject(pluginCommandObj, customCommand);
-
-    return this.checkPluginPath(pluginCommandObj, customCommand);
-  },
 
-  // ensures third-party command is actually supported by configured plugins
-  checkPluginObject(pluginCommandObj, customCommand) {
-    if (Object.keys(pluginCommandObj).length === 0)
-      throw new TruffleError(`\nError: "${customCommand}" command not supported by any currently configured plugins. Please make sure:
-    – plugins are correctly configured in truffle-config.js
-    – the plugin supporting the command you want to use is installed\n`);
-  },
-
-  // enforces relative command paths
-  checkPluginPath(pluginCommandObj, customCommand) {
-    const pluginModule = Object.keys(pluginCommandObj)[0];
-    const relativeCommandPath = Object.values(pluginCommandObj)[0];
-
-    if (path.isAbsolute(relativeCommandPath))
-      throw new TruffleError(
-        `\nError: Absolute paths not allowed!\nPlease change the '${customCommand}' path in ${pluginModule}/truffle-plugin.json to a relative path.\n`
-      );
-
-    return this.requirePlugin(pluginModule, relativeCommandPath);
-  },
-
-  // resolves and requires the plugin script
-  requirePlugin(pluginModule, relativeCommandPath) {
-    let pluginPath = originalRequire.resolve(pluginModule);
-    pluginPath = path.dirname(pluginPath);
-    pluginPath = path.resolve(pluginPath, relativeCommandPath);
-
-    const commandModule = originalRequire(pluginPath);
-    return commandModule;
+    // customCommand not found or failed to load
+    throw new TruffleError(
+      `\nError: "${customCommand}" command not supported by any currently configured plugins. Please make sure:
+  – plugins are correctly configured in truffle-config.js
+  – the plugin supporting the command you want to use is installed\n`
+    );
   },
 
   // executes command or throws user helpful error
-  run(pluginConfigs, customCommand, config, done) {
-    const runCommand = this.initializeCommand(pluginConfigs, customCommand);
+  run(customCommand, config, done) {
```

Comment:
Could we drop `done` in this method? Or do we need to keep it to support plugins written callback-style?